src/knx/bits.cpp

```diff
@@ -124,21 +124,19 @@ uint64_t sixBytesToUInt64(uint8_t* data)
 
 uint16_t crc16Ccitt(uint8_t* input, uint16_t length)
 {
-        uint32_t polynom = 0x1021;
-        uint8_t padded[length+2];
-
-        memcpy(padded, input, length);
-        memset(padded+length, 0x00, 2);
-
-        uint32_t result = 0xffff;
-        for (uint32_t i = 0; i < 8 * (uint32_t)sizeof(padded); i++) {
-            result <<= 1;
-            uint32_t nextBit = (padded[i / 8] >> (7 - (i % 8))) & 0x1;
-            result |= nextBit;
-            if ((result & 0x10000) != 0)
-                result ^= polynom;
-        }
-        return result & 0xffff;
+    uint32_t polynom = 0x1021;
+
+    uint32_t result = 0xffff;
+    for (uint32_t i = 0; i < 8 * ((uint32_t)length + 2); i++)
```

Comment:
You process bytes behind the length supplied as a parameter. I don't think that this should be done. I think the caller should provide a padded buffer and add the +2 to length if needed on the caller side.

Reply:
No, I don't - see my comment on Line 134.